cluster-api-gcp/util/retry.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"time"
+
+	backoff "github.com/cenkalti/backoff"
+)
+
+func NewExponentialBackoff(timeout time.Duration, maxInterval time.Duration) *backoff.ExponentialBackOff {
```

Comment:
(idea)
Have you considered having a  function signature more like ExponentialBackoff(timeout, maxInterval, function) error which would run the function in a backoff manner with the given timeout and maxInterval?

It would save code and this would be the only file that needs to import backoff.

Reply:
I thought about it, but it wouldn't really save much code - the func would still need to be where it is.

The reason why I didn't do is because what happens in func() could depend on other local/global methods that I'm not sure can be called from a different package. Example, `d.writeConfigToDisk(config)` is called in the retry function in `copyKubeConfig`, but I'm not sure if the scope of `d` is available anymore, and if that's true for any such cases that might be implemented in the future.

Reply:
So long as d is in scope where you define the func, I believe it should work. Have you found it not to work?